tests/experimental/nn/test_sync_batchnorm.py

```diff
@@ -92,6 +93,19 @@ def parity3d_bn():
     check_parity(torch_bn, fs_bn, x)
 
 
+@pg_test()
+def parity3d_checkpoint_syncbn():
+    rank = dist.get_rank()
+    torch.cuda.set_device(rank)
+    torch.manual_seed(rank)
+
+    x = torch.randn(4, 3, 4, 4, 4).cuda() * rank
+    torch_bn = torch.nn.SyncBatchNorm(3).cuda()
+    fs_bn = SyncBatchNorm(3).cuda()
+    fs_bn = checkpoint_wrapper(fs_bn, maintain_forward_counter=True)
+    check_parity_ddp(torch_bn, fs_bn, x)
```

Comment:
does check parity function actually check the buffer values? I think the running stats are kept in the buffers and they are not used during training and only later get used during inference? Sorry if I misunderstood that part.

Reply:
check_parity_ddp checks that the buffer values match:

https://github.com/facebookresearch/fairscale/blob/d240b748546445374e3823b4f8c3557963ae77f5/tests/experimental/nn/test_sync_batchnorm.py#L74